mutt_str_replace causes SIGSEGV if input and output reference the same string.
Fixes 31073116.

diff -u -p -r mutt-1.13.4.orig/lib.c mutt-1.13.4/lib.c
--- mutt-1.13.4.orig/lib.c	2020-03-24 15:17:33.418230443 +0000
+++ mutt-1.13.4/lib.c	2020-03-24 15:22:56.093810525 +0000
@@ -283,8 +283,9 @@ char *safe_strncat (char *d, size_t l, c
 
 void mutt_str_replace (char **p, const char *s)
 {
-  FREE (p);		/* __FREE_CHECKED__ */
+  char *tmp = *p;
   *p = safe_strdup (s);
+  FREE (&tmp);		/* __FREE_CHECKED__ */
 }
 
 void mutt_str_adjust (char **p)
